flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputGate.java

```diff
@@ -78,32 +79,32 @@
 
 	public abstract String getOwningTaskName();
 
-	public abstract boolean isFinished();
-
 	public abstract void requestPartitions() throws IOException, InterruptedException;
 
 	/**
 	 * Blocking call waiting for next {@link BufferOrEvent}.
 	 *
 	 * @return {@code Optional.empty()} if {@link #isFinished()} returns true.
 	 */
-	public abstract Optional<BufferOrEvent> getNextBufferOrEvent() throws IOException, InterruptedException;
+	public abstract Optional<BufferOrEvent> getNext() throws IOException, InterruptedException;
```

Comment:
Even through get a buffer or event without blocking, it still can be named with getNextBufferOrEvent(). What do you think?